spec/support/mailer_helpers.rb

```diff
@@ -7,7 +7,7 @@ def delivered_emails
     ActionMailer::Base.deliveries
   end
 
-  def devise_token_from_last_mail(token_name)
-    last_email.body.to_s[/#{token_name}_token=([^)]*)/, 1]
+  def first_link_from_last_mail
```

Comment:
For now this works because the first link is always what we need in the test, which might still be the case in the future as these emails are pretty standard. But the previous method was specific in finding the specific token you need.

Reply:
That's true. I think there is a slight benefit in replacing a test that the email contains the correct token, with a test that the email contains the correct link: the token could be inserted into the wrong link.